### ISSUE TITLE
Discourage the use of `grant*` methods

### ISSUE THREAD

**Issue by otaviomacedo** [ren-yamanashi/eslint-cdk-plugin#342]
### Feature Description

The new rule should check for the use of `grant*` methods in L2s that have a public `grants` attribute, and guide the user to use the new idiom: `foo.grants.someAction` instead of `foo.grantSomeAction`.

### Motivation

Historically, L2s that represent resources that can be accessed have `grant*` methods. More recently, a new pattern was introduced: Grants classes. These are auxiliary classes for resources, that contain all the grant methods. For example, for SNS topics, there is now a `TopicGrants`, with methods `publish` and `subscribe`, containing the logic that used to be in `Topic.grantPublish` and `Topic.grantSubscribe`, respectively.

The goal of these Grant classes is to allow the assignment of permissions to both L1s and L2s in a uniform way:

```ts
// With L2s:
const topic = new Topic(stack, ...);
TopicGrants.fromTopic(topic).subscribe(role);

// With L1s:
const topic = new CfnTopic(stack, ...);
TopicGrants.fromTopic(topic).subscribe(role);
```

For convenience, an instance of the auxiliary Grants class is exposed by the L2 constructs via the `grants` property:

```ts
const topic = new Topic(stack, ...);
topic.grants.subscribe(role);
```

This is the preferred idiom, instead of the old `grant*` methods. Although its usage is discouraged, these methods are not formally deprecated, to avoid too much noise for current users.

### Invalid Code Examples

```ts
const topic = new Topic(stack, ...);
topic.grantsSubscribe(role);
```

### Valid Code Examples

```ts
const topic = new Topic(stack, ...);
topic.grants.subscribe(role);
```

### Rule Type

suggestion (improve code)

### Implementation Details

- [x] This rule is related to AWS CDK best practices
- [x] I'm willing to submit a PR for this feature
